File: LC2215.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        map<int, int> m1, m2;
        vector <vector<int>> res;

        for(auto i : nums1)
            m1[i] ++;

        for(auto i : nums2)
            m2[i]++;
        
        vector<int> v;
        for(auto i : m1){
            if(m2.find(i.first) == m2.end())
                v.push_back(i.first);
        }
        res.push_back(v);
        v.clear();

        for(auto i : m2)
            if(m1.find(i.first) == m1.end())
                v.push_back(i.first);
        
        res.push_back(v);

        return res;
    }
};

File: LC2215.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        return [set(nums1) - set(nums2), set(nums2) - set(nums1)]


# Another method

class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        s1 = set(nums1)
        s2 = set(nums2)
        res = []
        l = []

        for i in s1:
            if i not in s2:
                l.append(i)
        
        res.append(l)
        l = []

        for i in s2:
            if i not in s1:
                l.append(i)
        
        res.append(l)

        return res
